http-native/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -1621,139 +1547,183 @@ private static String getServerName() {
+        List<String> sslEnabledProtocolsValueList = Arrays.asList(
+                protocol.getArrayValue(SECURESOCKET_CONFIG_PROTOCOL_VERSIONS).getStringArray());
+        if (!sslEnabledProtocolsValueList.isEmpty()) {
+            String sslEnabledProtocols = sslEnabledProtocolsValueList.stream().collect(Collectors.joining(",", "", ""));
+            Parameter serverProtocols = new Parameter(ANN_CONFIG_ATTR_SSL_ENABLED_PROTOCOLS, sslEnabledProtocols);
+            paramList.add(serverProtocols);
         }
-        if (ocspStapling != null) {
-            boolean ocspStaplingEnabled = ocspStapling.getBooleanValue(HttpConstants.ENABLE);
-            listenerConfiguration.setOcspStaplingEnabled(ocspStaplingEnabled);
-            long cacheSize = ocspStapling.getIntValue(HttpConstants.SSL_CONFIG_CACHE_SIZE);
-            long cacheValidationPeriod = ocspStapling.getIntValue(HttpConstants.SSL_CONFIG_CACHE_VALIDITY_PERIOD);
-            listenerConfiguration.setValidateCertEnabled(ocspStaplingEnabled);
-            if (ocspStaplingEnabled) {
-                if (cacheSize != 0) {
-                    listenerConfiguration.setCacheSize(Math.toIntExact(cacheSize));
-                }
-                if (cacheValidationPeriod != 0) {
-                    listenerConfiguration.setCacheValidityPeriod(Math.toIntExact(cacheValidationPeriod));
-                }
-            }
+        String sslProtocol = protocol.getStringValue(SECURESOCKET_CONFIG_PROTOCOL_NAME).getValue();
+        if (!sslProtocol.isBlank()) {
+            sslConfiguration.setSSLProtocol(sslProtocol);
         }
-        listenerConfiguration.setTLSStoreType(PKCS_STORE_TYPE);
-        String serverEnableSessionCreation = String
-                .valueOf(sslConfig.getBooleanValue(SSL_CONFIG_ENABLE_SESSION_CREATION));
-        Parameter enableSessionCreationParam = new Parameter(SSL_CONFIG_ENABLE_SESSION_CREATION.getValue(),
-                                                             serverEnableSessionCreation);
-        serverParamList.add(enableSessionCreationParam);
-        if (!serverParamList.isEmpty()) {
-            listenerConfiguration.setParameters(serverParamList);
+    }
+
+    private static void evaluateCertValidationField(BMap<BString, Object> certValidation,
+                                                    SslConfiguration sslConfiguration) {
+        String type = certValidation.getStringValue(SECURESOCKET_CONFIG_CERT_VALIDATION_TYPE).getValue();
+        if (type.equals(SECURESOCKET_CONFIG_CERT_VALIDATION_TYPE_OCSP_STAPLING.getValue())) {
+            sslConfiguration.setOcspStaplingEnabled(true);
+        }
+        long cacheSize = certValidation.getIntValue(SECURESOCKET_CONFIG_CERT_VALIDATION_CACHE_SIZE).intValue();
+        long cacheValidityPeriod = ((BDecimal) certValidation.get(
+                SECURESOCKET_CONFIG_CERT_VALIDATION_CACHE_VALIDITY_PERIOD)).intValue();
+        sslConfiguration.setValidateCertEnabled(true);
```

Comment:
Shall we move the above line to the else part of `if (type.equals(SECURESOCKET_CONFIG_CERT_VALIDATION_TYPE_OCSP_STAPLING.getValue())) {`

?

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerina-http/pull/255/commits/1a7dc49ebf025aeddaec010e193c4e184353d2bb